Published-source: https://deb.freexian.com/extended-lts/pool/main/p/php7.0/php7.0_7.0.33-0+deb9u26.debian.tar.xz
Published-patch: 0126-Fix-bug-79330-make-all-execution-modes-consistent-in.patch
Scope: selected PHP runtime components and their tests; packaging and unrelated extensions omitted.

From: Stanislav Malyshev <stas@php.net>
Date: Mon, 13 Apr 2020 21:00:44 -0700
Subject: Fix bug #79330 - make all execution modes consistent in rejecting \0

(cherry picked from commit 14fcc813948254b84f382ff537247d8a7e5e0e62)
---
 ext/standard/exec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/ext/standard/exec.c
+++ b/ext/standard/exec.c
@@ -524,6 +524,15 @@ PHP_FUNCTION(shell_exec)
 		return;
 	}
 
+	if (!command_len) {
+		php_error_docref(NULL, E_WARNING, "Cannot execute a blank command");
+		RETURN_FALSE;
+	}
+	if (strlen(command) != command_len) {
+		php_error_docref(NULL, E_WARNING, "NULL byte detected. Possible attack");
+		RETURN_FALSE;
+	}
+
 #ifdef PHP_WIN32
 	if ((in=VCWD_POPEN(command, "rt"))==NULL) {
 #else
